### PR TITLE
Backend: Add sqlalchemy database models

### PR DESCRIPTION
**Pull request by OxygenCobalt**
Required db module to be split off to avoid ciruclar dependency.

### CODE REVIEW

**Comment by grantlemons** [comment]
you're using declarative base, but the imperative style mapping as defined in the docs

https://docs.sqlalchemy.org/en/20/orm/mapping_styles.html#declarative-mapping

if it works, I guess that's fine, but if our project wants to stick to being totally idiomatic we might want to use declarative in it's entirety.

File: backend/src/model/alert.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from src.db import Base
+
+
+class AlertModel(Base):
+    __tablename__ = "alerts"
+    id: Mapped[int] = mapped_column(
+        primary_key=True, autoincrement=True, nullable=False
+    )
+    text: Mapped[str] = mapped_column(nullable=False)
+    start_datetime = mapped_column(nullable=False)
```
Comment (tbwrigh) [inline backend/src/model/alert.py:13]: Why is start_datetime not typed?

File: backend/src/model/route.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+from sqlalchemy.orm import Mapped, mapped_column
+from src.db import Base
+
+
+class RouteModel(Base):
+    __tablename__ = "routes"
+    id: Mapped[int] = mapped_column(
+        primary_key=True, autoincrement=True, nullable=False
+    )
+    name: Mapped[str] = mapped_column(unique=True)
```
Comment (tbwrigh) [inline backend/src/model/route.py:11]: Is there a reason this isn't marked `nullable=false`

File: backend/src/model/route_van.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from sqlalchemy import ForeignKeyConstraint
+from sqlalchemy.orm import Mapped, mapped_column
+from src.db import Base
+
+
+class RouteVanModel(Base):
+    __tablename__ = "route_vans"
```
Comment (tbwrigh) [inline backend/src/model/route_van.py:7]: Looking in alembic, I don't see a table named route_vans.

File: backend/src/model/stop.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class StopModel(Base):
+    __tablename__ = "stops"
+    id: Mapped[int] = mapped_column(
+        primary_key=True, autoincrement=True, nullable=False
+    )
+    name: Mapped[str] = mapped_column(unique=True, nullable=False)
+    lat: Mapped[float] = mapped_column(nullable=False)
+    lon: Mapped[float] = mapped_column(nullable=False)
```
Comment (tbwrigh) [inline backend/src/model/stop.py:12]: Should there be something to indicate the table property `UNIQUE(lat, lon)`?

File: backend/src/model/van.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+from sqlalchemy.orm import Mapped, mapped_column
+from src.db import Base
+
+
+class VanModel(Base):
+    __tablename__ = "vans"
+    id: Mapped[int] = mapped_column(
+        primary_key=True, autoincrement=True, nullable=False
+    )
+    wheelchair: Mapped[bool] = mapped_column(nullable=False)
```
Comment (tbwrigh) [inline backend/src/model/van.py:11]: In alembic I see this table as having route_id int it.

File: backend/src/db.py
Comment (tbwrigh) [inline backend/src/db.py:1]: perhaps we should just go ahead and bite the bullet and make the connection pool in this file/pr?

**Review by tbwrigh — CHANGES_REQUESTED** [review]

File: backend/src/model/route.py (hunk 2 above)
Comment (OxygenCobalt) [reply backend/src/model/route.py:11]: Copilot hallucination

File: backend/src/model/alert.py (hunk 1 above)
Comment (OxygenCobalt) [reply backend/src/model/alert.py:13]: Mistake with sed

File: backend/src/model/route_van.py (hunk 3 above)
Comment (OxygenCobalt) [reply backend/src/model/route_van.py:7]: Seems like I was working with an older copy of the alembic tables

File: backend/src/model/van.py (hunk 5 above)
Comment (OxygenCobalt) [reply backend/src/model/van.py:11]: Used old copies of alembic tables, it seems

File: backend/src/db.py
Comment (grantlemons) [reply backend/src/db.py:1]: I'd say make a separate pr if only for the principal of it

**Comment by grantlemons** [comment]
will review after tyler's requested changes are resolved

File: backend/src/db.py
Comment (tbwrigh) [reply backend/src/db.py:1]: sure

File: backend/src/model/route_van.py (hunk 3 above)
Comment (tbwrigh) [reply backend/src/model/route_van.py]: Why'd you mark this as resolved? Something definitely needs to change here.

**Pull request merged**